api/bucketsd/service.go

```diff
@@ -511,7 +512,10 @@ func (s *Service) addPinnedBytes(ctx context.Context, delta int64) context.Conte
 	owner, ok := buckets.BucketOwnerFromContext(ctx)
 	if ok {
 		owner.StorageUsed += delta
-		owner.StorageAvailable -= delta
+		// int64(math.MaxInt64) indicates that the user has no current cap so don't deduct
+		if owner.StorageAvailable < int64(math.MaxInt64) {
+			owner.StorageAvailable -= delta
```

Comment:
seems logical to not try and decrease storage available if the user has uncapped storage

Reply:
Oh, didn't know `int64(math.MaxInt64)` exactly mean uncapped.